packages/swapper/src/swappers/JupiterSwapper/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,231 @@
+    commissionBps: affiliateBps,
+    amount: sellAmount,
+    slippageBps: convertDecimalPercentageToBasisPoints(
+      slippageTolerancePercentageDecimal ??
+        getDefaultSlippageDecimalPercentageForSwapper(SwapperName.Jupiter),
+    ).toFixed(),
+  })
+
+  if (maybeQuoteResponse.isErr()) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Quote request failed',
+        code: TradeQuoteError.NoRouteFound,
+      }),
+    )
+  }
+
+  const { data: quoteResponse } = maybeQuoteResponse.unwrap()
+
+  const getFeeData = async () => {
+    const { chainNamespace } = fromAssetId(sellAsset.assetId)
+
+    switch (chainNamespace) {
+      case CHAIN_NAMESPACE.Solana: {
+        const sellAdapter = deps.assertGetSolanaChainAdapter(sellAsset.chainId)
+        const getFeeDataInput: GetFeeDataInput<KnownChainIds.SolanaMainnet> = {
+          // Simulates a self-send, since we don't know the to just yet at this stage
+          to: input.sendAddress!,
+          value: sellAmount,
+          chainSpecific: {
+            from: input.sendAddress!,
+            tokenId:
+              sellAsset.assetId === solAssetId
+                ? undefined
+                : fromAssetId(sellAsset.assetId).assetReference,
+          },
+        }
+        const { fast } = await sellAdapter.getFeeData(getFeeDataInput)
+        return { networkFeeCryptoBaseUnit: fast.txFee }
+      }
+
+      default:
+        throw new Error('Unsupported chainNamespace')
+    }
+  }
+
+  const protocolFees: Record<AssetId, ProtocolFee> = quoteResponse.routePlan.reduce(
+    (acc, route) => {
+      const feeAssetId = toAssetId({
+        assetReference: route.swapInfo.feeMint,
+        assetNamespace: ASSET_NAMESPACE.splToken,
+        chainNamespace: CHAIN_NAMESPACE.Solana,
+        chainReference: CHAIN_REFERENCE.SolanaMainnet,
+      })
+      const feeAsset = assetsById[feeAssetId]
+
+      // If we can't find the feeAsset, we can't provide a protocol fee to display
+      // But these fees exists at protocol level, it's mostly to make TS happy as we should have the market data and assets
+      if (!feeAsset) return acc
+
+      acc[feeAssetId] = {
+        requiresBalance: false,
+        amountCryptoBaseUnit: bnOrZero(route.swapInfo.feeAmount).toString(),
+        asset: feeAsset,
+      }
+
+      return acc
+    },
+    {} as Record<AssetId, ProtocolFee>,
+  )
+
+  const getQuoteRate = (sellAmountCryptoBaseUnit: string, buyAmountCryptoBaseUnit: string) => {
+    return getRate({
+      sellAmountCryptoBaseUnit,
+      buyAmountCryptoBaseUnit,
+      sellAsset,
+      buyAsset,
+    })
+  }
+
+  const quotes: TradeQuote[] = []
+
+  const feeData = await getFeeData()
+
+  const rate = getQuoteRate(quoteResponse.inAmount, quoteResponse.outAmount)
+
+  const tradeQuote: TradeQuote = {
+    id: uuid(),
+    rate,
+    receiveAddress,
+    potentialAffiliateBps: affiliateBps,
+    affiliateBps,
+    isStreaming: false,
```

Comment:
nitpick: this is optional and not applicable to this domain so might as well remove this line